From patchwork Mon Oct  8 14:41:38 2018
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Wei Yang <richard.weiyang@gmail.com>
X-Patchwork-Id: 10630897
Return-Path: <kvm-owner@kernel.org>
Received: from mail.wl.linuxfoundation.org (pdx-wl-mail.web.codeaurora.org
 [172.30.200.125])
	by pdx-korg-patchwork-2.web.codeaurora.org (Postfix) with ESMTP id 1281613BB
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  8 Oct 2018 14:41:50 +0000 (UTC)
Received: from mail.wl.linuxfoundation.org (localhost [127.0.0.1])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 01EC42955A
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  8 Oct 2018 14:41:50 +0000 (UTC)
Received: by mail.wl.linuxfoundation.org (Postfix, from userid 486)
	id E87A32955F; Mon,  8 Oct 2018 14:41:49 +0000 (UTC)
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on
	pdx-wl-mail.web.codeaurora.org
X-Spam-Level: 
X-Spam-Status: No, score=-8.0 required=2.0 tests=BAYES_00,DKIM_SIGNED,
	DKIM_VALID,DKIM_VALID_AU,FREEMAIL_FROM,MAILING_LIST_MULTI,RCVD_IN_DNSWL_HI
	autolearn=ham version=3.3.1
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by mail.wl.linuxfoundation.org (Postfix) with ESMTP id 8CD1C2955A
	for <patchwork-kvm@patchwork.kernel.org>;
 Mon,  8 Oct 2018 14:41:49 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S1726439AbeJHVxv (ORCPT
        <rfc822;patchwork-kvm@patchwork.kernel.org>);
        Mon, 8 Oct 2018 17:53:51 -0400
Received: from mail-pf1-f194.google.com ([209.85.210.194]:37100 "EHLO
        mail-pf1-f194.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S1726291AbeJHVxv (ORCPT <rfc822;kvm@vger.kernel.org>);
        Mon, 8 Oct 2018 17:53:51 -0400
Received: by mail-pf1-f194.google.com with SMTP id j23-v6so8483646pfi.4
        for <kvm@vger.kernel.org>; Mon, 08 Oct 2018 07:41:48 -0700 (PDT)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=gmail.com; s=20161025;
        h=from:to:cc:subject:date:message-id;
        bh=JqV/Ec00XEPyciKcGWcxPLunmj0aB4O8xHRzldeW13g=;
        b=B3kMWmbxER/LP2SiYXfQzUZhfQ0XJDTpgvAPBQRDS5TeNtRO8JVZ/W1qeTqRgyfamv
         EiUKqxvuzDCDm1cD7JuhWaLERv4ZtcNcvTJW5OQ0wu7Winbu8FzNE2jXPHGwEtUWm2c1
         s22HGOS2sMvYJKlTuv+jQwk49FfZmsuBxJBaMSCqNZIg2Bw6p4YYtVzi5/jtK5xUkUOt
         h3TvDI2OMTnMq0Avt7xpfJhCgqcxozzJP9/+GU+4Wf7+dWpbRWFy8yqvU3DS9nvuC/0x
         YRcF3whROHl2L5lL18iqDSixKZdccapjIEsuapOWmwGJtNmKGtmrgrKH3mp8vwdjkNt6
         GPBg==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20161025;
        h=x-gm-message-state:from:to:cc:subject:date:message-id;
        bh=JqV/Ec00XEPyciKcGWcxPLunmj0aB4O8xHRzldeW13g=;
        b=kcYsoKOcsibgS+RikGuXKSBOZrVQf+CoCVYTpucDq63ZeOCVtqNuUOubYSAN4yxhgv
         J1bzUe8Jel3Z8Y1q5nfSVdIVJfW/nGcDkftp858k5L0RRvu3syjYzomGHr0wxeFOfFmD
         hiLavKaJt/j0EN2uWKAVVp4oMZEOmP4U7fEYwxpx+S6+H+vObiww0gef5uyF5s/zj+gq
         ztqjQIwcQHq4qipX+Dhon0F1IzxVZp3X7YwLw7rPVzvc0duf7ql4c6GlGzmrxexPxsHo
         Tmk6tek1GyMcPW/GQTAfNSIFha47sBdfMS1l/eyT9+9HwAKgETg1NGHF7QplEwtWYu4t
         wSgA==
X-Gm-Message-State: ABuFfoi0qN759EwV1X2VAeQEnDy+hatV4gTfpZfcdWSP1tpd1a7nSjt9
        Jfhu/ZO4kHyE6kw9/neNOek=
X-Google-Smtp-Source: 
 ACcGV62Nm5gWEjVDY7X9pzkaQ0MMSOrSSlm1lc+BHagKdXZAZo+BKobYWNIEd4Xf/RxZH7H+d0C6MA==
X-Received: by 2002:a63:5b57:: with SMTP id
 l23-v6mr5330666pgm.50.1539009707681;
        Mon, 08 Oct 2018 07:41:47 -0700 (PDT)
Received: from localhost ([185.92.221.13])
        by smtp.gmail.com with ESMTPSA id
 q76-v6sm29276118pfa.18.2018.10.08.07.41.46
        (version=TLS1_2 cipher=ECDHE-RSA-AES128-GCM-SHA256 bits=128/128);
        Mon, 08 Oct 2018 07:41:47 -0700 (PDT)
From: Wei Yang <richard.weiyang@gmail.com>
To: pbonzini@redhat.com, rkrcmar@redhat.com
Cc: kvm@vger.kernel.org, Wei Yang <richard.weiyang@gmail.com>
Subject: [RFC PATCH] KVM: remove the writable page for read fault case in
 hva_to_pfn_slow()
Date: Mon,  8 Oct 2018 22:41:38 +0800
Message-Id: <20181008144138.41677-1-richard.weiyang@gmail.com>
X-Mailer: git-send-email 2.15.1
Sender: kvm-owner@vger.kernel.org
Precedence: bulk
List-ID: <kvm.vger.kernel.org>
X-Mailing-List: kvm@vger.kernel.org
X-Virus-Scanned: ClamAV using ClamSMTP

Case (!write_fault && writable) has been handled in hva_to_pfn_fast(),
it is not necessary to try again if hva_to_pfn_fast() already failed.

This patch removes this case in hva_to_pfn_slow().

Signed-off-by: Wei Yang <richard.weiyang@gmail.com>
---

Hope my understanding is correct.

---
 virt/kvm/kvm_main.c | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/virt/kvm/kvm_main.c b/virt/kvm/kvm_main.c
index 1f42f1d474b5..c8fb3a9d81fa 100644
--- a/virt/kvm/kvm_main.c
+++ b/virt/kvm/kvm_main.c
@@ -1403,16 +1403,6 @@ static int hva_to_pfn_slow(unsigned long addr, bool *async, bool write_fault,
 	if (npages != 1)
 		return npages;
 
-	/* map read fault as writable if possible */
-	if (unlikely(!write_fault) && writable) {
-		struct page *wpage;
-
-		if (__get_user_pages_fast(addr, 1, 1, &wpage) == 1) {
-			*writable = true;
-			put_page(page);
-			page = wpage;
-		}
-	}
 	*pfn = page_to_pfn(page);
 	return npages;
 }
